nasl/nasl_packet_forgery.c

```diff
@@ -570,9 +570,9 @@ get_tcp_element (lex_ctxt *lexic)
   else if (!strcmp (element, "data"))
     {
       retc = alloc_typed_cell (CONST_DATA);
-      retc->size = UNFIX (ip->ip_len) - ntohl (tcp->th_off) * 4;
+      retc->size = UNFIX (ip->ip_len) - (tcp->th_off + ip->ip_hl) * 4;
```

Comment:
I guess a similar fix for this done in 3e1a3db295f0b649bdef22868fdb148513e9f2af is required for `get_tcp_v6_element` here as well?

https://github.com/greenbone/openvas/blob/d589114edb38c082e5ee25b9d364f24a3a8cc34d/nasl/nasl_packet_forgery_v6.c#L592

At least when getting the data element with `get_tcp_v6_element()` (see internal tracker for a reproducer) i'm getting the following message:

```
lib  nasl-Message: 13:51:28.877: [70564](tcp_data.nasl:59) get_tcp_v6_element: Erroneous tcp header offset 0
```

Reply:
I think the size calculation is correct in this case. But data of size 0 was considered as an error. So I think it was just a wrong error message. Should be fixed now.

Reply:
Indeed this should do the trick. The "header offset 0" was a little bit misleading in this case.